freeipa/src/main/java/com/sequenceiq/freeipa/sync/FreeipaStatusInfoLogger.java

```diff
@@ -0,0 +1,36 @@
+package com.sequenceiq.freeipa.sync;
+
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class FreeipaStatusInfoLogger {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeipaStatusInfoLogger.class);
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    public void logFreeipaStatus(Stack stack, Set<InstanceMetaData> checkableInstances) {
```

Comment:
I would suggest some refactor:
```
    public void logFreeipaStatus(Stack stack, Set<InstanceMetaData> checkableInstances) {
        String freeipaStatusInfo = String.format("freeipa %s is %s.", stack.getResourceCrn(), stack.getStackStatus().getStatus());
        String allInstanceStatusInfo = createInstancesStatusInfo(checkableInstances);
        LOGGER.debug(":::Auto sync::: freeipa status from healtch check: {} {}", freeipaStatusInfo, allInstanceStatusInfo);
    }

    private String createInstancesStatusInfo(Set<InstanceMetaData> checkableInstances) {
        Set<InstanceMetaData> allInstanceMetaData = fetchCheckableInstancesFromDb(checkableInstances);
        String allInstanceStatusInfo = allInstanceMetaData.stream().map(instance -> String.format("%s is %s", instance.getInstanceId(),
                instance.getInstanceStatus())).collect(Collectors.joining("; "));
        return allInstanceStatusInfo;
    }

    private Set<InstanceMetaData> fetchCheckableInstancesFromDb(Set<InstanceMetaData> checkableInstances) {
        Set<String> checkableInstanceIds = collectCheckableInstanceIds(checkableInstances);
        Set<InstanceMetaData> allInstanceMetaData = instanceMetaDataService
                .getByInstanceIds(checkableInstanceIds);
        return allInstanceMetaData;
    }

    private Set<String> collectCheckableInstanceIds(Set<InstanceMetaData> checkableInstances) {
        return checkableInstances.stream()
                .map(InstanceMetaData::getInstanceId)
                .collect(Collectors.toSet());
    }
```

the main just next to breaking it into smaller parts that I have switched to stream and used the built in collector instead of creating a StringBuilder. Also I modified the log line and removed the trailing space from `freeipaStatusInfo`. It's a bit odd to add a space there.
Feel free to modify the delimiter in joining, I just added something.
Please also note I haven't added the fix for the Stack status

Reply:
Thanks a lot! I refactored and also added the fix for stack status.

Reply:
Post the new log message here.  

2020-12-08 11:33:03,669 [quartzScheduler_Worker-1] logFreeipaStatus:31 DEBUG c.s.f.s.FreeipaStatusInfoLogger - [type:STACK] [crn:crn:cdp:freeipa:us-west-1:default:freeipa:e09e4a42-07f6-4298-a10f-6e031ab8217f] [name:yuwang-env2-15433-freeipa] [flow:] [requestid:c4ec4dfb-827f-4975-a5d1-b70ceda14636] [tenant:default] [userCrn:] [environment:crn:cdp:environments:us-west-1:default:environment:6298663a-49b4-43b2-97d7-c2bd74e67f77] [traceId:] [spanId:] [operationId:] :::Auto sync::: freeipa status from healtch check: freeipa stack is AVAILABLE.  instance i-09ecf6806dd902fcc is CREATED; instance i-078af5be4d2eece5d is CREATED